lib/enum/extra.ex

```diff
@@ -322,4 +324,140 @@ defmodule Enum.Extra do
   def find_or_error(enum, default \\ :error, fun) when is_function(fun, 1) do
     Enum.find_value(enum, default, fn x -> if fun.(x), do: {:ok, x} end)
   end
+
+
+  @doc """
+  Returns the enumerable from the `list_of_enums` which contains the most items.
+
+  If there is a tie an enum earlier in `list_of_enums` will take precedence.
+
+  Note: This function will count items in all the enums until there is only one left that has not
+  halted. Using this with two infinite enums will result in this function never returning.
+
+      iex> Enum.Extra.longest([[1], [2, 3], [4, 5, 6]])
+      [4, 5, 6]
+
+      iex> Enum.Extra.longest([[1], [2]])
+      [1]
+
+  """
+  @spec longest([Enum.t]) :: Enum.t
+  def longest(list_of_enums) when is_list(list_of_enums) and length(list_of_enums) > 0 do
+    {longest_quick_count, slow_counts} =
+      partition_longest_quick_counts(list_of_enums)
+
+    {longest_iter, slow_count} =
+      slow_counts
+      |> :lists.reverse()
+      |> Enum.map(&Iterator.new/1)
+      |> do_longest(0)
+
+    case longest_quick_count do
+      {_quick_enum, quick_count} when quick_count <  slow_count ->
+        longest_iter.enum
```

Comment:
do we need to call `enum` here? Isn't an `Iterator` an `Enum.t` anyway?

Reply:
Yes, but returning the same value as the input is the point of this function. ie. What if the input implements other protocols? It'd be annoying to lose that, nor have any information as to which input enumerable won out.

Reply:
oh, sure, that seems reasonable